snapcraft/internal/os_release.py

```diff
@@ -38,12 +38,16 @@ def __init__(self, *, os_release_file: str = '/etc/os-release') -> None:
 
         :param str os_release_file: Path to os-release file to be parsed.
         """
-        with open(os_release_file) as f:
-            self._os_release = {}  # type: Dict[str, str]
-            for line in f:
-                entry = line.rstrip().split('=')
-                if len(entry) == 2:
-                    self._os_release[entry[0]] = entry[1].strip('"')
+        self._os_release = {}  # type: Dict[str, str]
+        try:
+            with open(os_release_file) as f:
+                for line in f:
+                    entry = line.rstrip().split('=')
+                    if len(entry) == 2:
+                        self._os_release[entry[0]] = entry[1].strip('"')
+        except FileNotFoundError:
+            self._os_release['ID'] = 'unknown'
+            self._os_release['NAME'] = 'unknown'
```

Comment:
The accessor methods are designed to raise errors if the ID or NAME aren't specified, so it's probably not necessary to set these to "unknown". Just don't set anything at all if the file doesn't exist and the rest _should_ work out.

Reply:
My goal was to have a fallback for the case where 1) `/etc/os-release` doesn't exist 2) `_is_deb_based` called from `repo/__init__.py` should return `False` 3) `_get_repo_for_platform` should fallback to the `DummyRepo`. The idea being, we could extend this in the future to populate the values from other sources.

That is, assuming we want `OsRelease` be our main entry point for OS delection. Otherwise we'd have to wrap this in a new API that's not based on a text file.

Reply:
Hmm. I think I'd rather see `_is_deb_based` and `_get_repo_for_platform` catch the `OsReleaseIdError` rather than make `OsRelease` work when there _is_ no `/etc/os-release`. It seems to dilute its purpose.